roofit/xroofit/src/xRooNode.cxx

```diff
@@ -31,6 +35,43 @@
 #include "RooSimultaneous.h"
 #undef protected
 
+#define GETWS(a) a->_myws
+#define GETWSSETS(w) w->_namedSets
+#define GETWSSNAPSHOTS(w) w->_snapshots
+#define GETACTBROWSER(b) b->fActBrowser
+#define GETROOTDIR(b) b->fRootDir
+#define GETLISTTREE(b) b->fListTree
+#define GETDMP(o,m) o->m
+
+#else
+
+#include "RooAbsArg.h"
+#include "RooWorkspace.h"
+#include "RooFitResult.h"
+#include "RooConstVar.h"
+#include "RooHistFunc.h"
+#include "RooRealSumPdf.h"
+#include "RooSimultaneous.h"
+#include "RooAddPdf.h"
+#include "RooProduct.h"
+#include "TPad.h"
+#include "RooStats/HistFactory/PiecewiseInterpolation.h"
+#include "RooStats/HistFactory/FlexibleInterpVar.h"
+#include "RooStats/HistFactory/ParamHistFunc.h"
+#include "RooProdPdf.h"
+#include "TRootBrowser.h"
+#include "TGFileBrowser.h"
+
+#define GETWS(a) a->workspace()
+#define GETWSSETS(w) w->sets()
+#define GETWSSNAPSHOTS(w) w->getSnapshots()
+#define GETACTBROWSER(b) b->GetActBrowser()
+#define GETROOTDIR(b) b->GetRootDir()
+#define GETLISTTREE(b) b->GetListTree()
```

Comment:
```suggestion
namespace {
RooWorkspace* GETWS(RooAbsArg * a) { return a->workspace(); }
const auto& GETWSSETS(RooWorkspace * w){ return  w->sets(); }
auto GETWSSNAPSHOTS(RooWorkspace * w){ return  w->getSnapshots(); }
auto GETACTBROWSER(TRootBrowser * b){ return  b->GetActBrowser(); }
auto GETROOTDIR(TGFileBrowser * b){ return  b->GetRootDir(); }
auto GETLISTTREE(TGFileBrowser * b) { return b->GetListTree(); }
} // namespace
```
It's better for type checking and safety to define actual functions rather than preprocessor macros.